dynamic_instrumentation/src/instrument.rs

```diff
@@ -323,13 +553,27 @@ impl<'tcx> Visitor<'tcx> for CollectInstrumentationPoints<'_, 'tcx> {
                         .arg_index_of(p.local)
                         .source(&source)
                         .dest(&dest)
+                        .instrumentation_priority(0)
                         .add_to(self);
                 };
             }
             _ => (),
         }
     }
 
+    fn visit_body(&mut self, body: &Body<'tcx>) {
+        // adds an instrumentation to mark the start of the body
+        self.super_body(body);
+        if self.instrumentation_points.len() > 0 {
+            let body_begin_func = self.hooks().find("mark_begin_body");
+            let start_loc = Location {
+                block: BasicBlock::from_u32(0),
+                statement_index: 0,
+            };
+            self.loc(start_loc, start_loc, body_begin_func).add_to(self);
+        }
```

Comment:
```suggestion
        
        if self.instrumentation_points.is_empty() {
            return;
        }
        let body_begin_func = self.hooks().find("mark_begin_body");
        let start_loc = Location::START;
        self.loc(start_loc, start_loc, body_begin_func).add_to(self);
```

`clippy::len_zero` warning.